File: SDKs/twilio.com/api/1.36.0/python/src/sdk/models/operations/listaccount.py
from dataclasses import dataclass, field
from typing import Enum,List,Optional
from dataclasses_json import dataclass_json
from sdk.models import shared
LIST_ACCOUNT_SERVERS = [
	"https://api.twilio.com",
]


@dataclass
class ListAccountQueryParams:
    friendly_name: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'FriendlyName', 'style': 'form', 'explode': True }})
    page_size: Optional[int] = field(default=None, metadata={'query_param': { 'field_name': 'PageSize', 'style': 'form', 'explode': True }})
    status: Optional[shared.AccountEnumStatusEnum] = field(default=None, metadata={'query_param': { 'field_name': 'Status', 'style': 'form', 'explode': True }})
    

@dataclass
class ListAccountSecurity:
    account_sid_auth_token: shared.SchemeAccountSidAuthToken = field(default=None, metadata={'security': { 'scheme': True, 'type': 'http', 'sub_type': 'basic' }})
    

@dataclass
class ListAccountRequest:
    server_url: Optional[str] = field(default=None)
    query_params: ListAccountQueryParams = field(default=None)
    security: ListAccountSecurity = field(default=None)
    

@dataclass_json
@dataclass
class ListAccountListAccountResponse:
    accounts: Optional[List[shared.APIV2010Account]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'accounts' }})
    end: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'end' }})
    first_page_uri: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'first_page_uri' }})
    next_page_uri: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'next_page_uri' }})
    page: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'page' }})
    page_size: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'page_size' }})
    previous_page_uri: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'previous_page_uri' }})
    start: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'start' }})
    uri: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'uri' }})
    

@dataclass
class ListAccountResponse:
    content_type: str = field(default=None)
    list_account_response: Optional[ListAccountListAccountResponse] = field(default=None)
    status_code: int = field(default=None)
    

File: SDKs/twilio.com/api/1.36.0/typescript/src/sdk/models/operations/listaccount.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
export const LISTACCOUNT_SERVERS = [
	"https://api.twilio.com",
];



export class ListAccountQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=FriendlyName" })
  friendlyName?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=PageSize" })
  pageSize?: number;

  @Metadata({ data: "queryParam, style=form;explode=true;name=Status" })
  status?: shared.AccountEnumStatusEnum;
}


export class ListAccountSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=http;subtype=basic" })
  accountSidAuthToken: shared.SchemeAccountSidAuthToken;
}


export class ListAccountRequest extends SpeakeasyBase {
  @Metadata()
  serverUrl?: string;

  @Metadata()
  queryParams: ListAccountQueryParams;

  @Metadata()
  security: ListAccountSecurity;
}


export class ListAccountListAccountResponse extends SpeakeasyBase {
  @Metadata({ data: "json, name=accounts", elemType: shared.ApiV2010Account })
  accounts?: shared.ApiV2010Account[];

  @Metadata({ data: "json, name=end" })
  end?: number;

  @Metadata({ data: "json, name=first_page_uri" })
  firstPageUri?: string;

  @Metadata({ data: "json, name=next_page_uri" })
  nextPageUri?: string;

  @Metadata({ data: "json, name=page" })
  page?: number;

  @Metadata({ data: "json, name=page_size" })
  pageSize?: number;

  @Metadata({ data: "json, name=previous_page_uri" })
  previousPageUri?: string;

  @Metadata({ data: "json, name=start" })
  start?: number;

  @Metadata({ data: "json, name=uri" })
  uri?: string;
}


export class ListAccountResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  listAccountResponse?: ListAccountListAccountResponse;

  @Metadata()
  statusCode: number;
}
